chemprop/cli/train.py

```diff
@@ -657,7 +662,11 @@ def main(args):
     monitor_mode = "min" if model.metrics[0].minimize else "max"
     logger.debug(f"Evaluation metric: '{model.metrics[0].alias}', mode: '{monitor_mode}'")
 
-    tb_logger = TensorBoardLogger(args.output_dir, "tb_logs")
+    try:
+        trainer_logger = TensorBoardLogger(args.output_dir, "trainer_logs")
+    except ImportError:
+        trainer_logger = CSVLogger(args.output_dir, "trainer_logs")
```

Comment:
the `ImportError` should be raised at the top of script when you try and import `TensorBoardLogger`, but looking the source,[^1][^2], it seems like it'll actually be a `ModuleNotFoundError`

[^1]: https://lightning.ai/docs/pytorch/stable/_modules/lightning/pytorch/loggers/tensorboard.html#TensorBoardLogger
[^2]: https://lightning.ai/docs/fabric/stable/_modules/lightning/fabric/loggers/tensorboard.html

Reply:
I'll change it to `ModuleNotFoundError`. For where to put the try-catch: Importing `TensorBoardLogger` isn't what raises the error. The error is raised when you try to instantiate a `TensorBoardLogger` object. If I were to put it at the top of the script it would need to look something like this:
```
from lightning import pytorch as pl
try:
	from lightning.pytorch.loggers import TensorBoardLogger as Logger
	Logger(Path())
except ModuleNotFoundError:
	from lightning.pytorch.loggers import CSVLogger as Logger
from lightning.pytorch.callbacks import ModelCheckpoint, EarlyStopping
import torch
[...]
trainer_logger = Logger(args.output_dir, "trainer_logs")
```
which I think looks clunky amid the imports, so leaving it down below is better.